RHIDP-1848: Updates the setting up RBAC permission (#906)
redhat-developer/red-hat-developers-documentation-rhdh: modules/authorization/proc-enabling-guest-access-for-rbac-frontend-plugin.adoc

== Configuring the RBAC backend plugin

Update the `app-config.yaml` to enable the permission framework, include the user `user:default/guest` in the admin users section, and ensure that the plugins with permissions appear in the UI as shown:

[source,yaml,subs=+quotes]
----
permission
  enabled: true
  rbac:
    admin:
      users:
        - name: user:default/guest
    pluginsWithPermission:
      - catalog
      - permission
      - scaffolder
----

[NOTE]
====
The `pluginsWithPermission` section includes only three plugins. Update the section as needed to include any additional plugins that also incorporate permissions.
====

== Setting up the guest authentication provider

To enable guest authentication and to use it alongside the RBAC frontend plugin, update the `app-config.yaml`. Add the user entity reference to resolve and enable the `dangerouslyAllowOutsideDevelopment` option, as shown in the following example:

[source,yaml,subs="+attributes,+quotes"]
----
auth:
  environment: development
  providers:
    guest:
      userEntityRef: user:default/guest
      dangerouslyAllowOutsideDevelopment: true
----

> You can use `user:default/guest` as the user entity reference to match the added user under the `permission.rbac.admin.users` section of the `app-config.yaml`.

Author: > Is this option or required? If optional, let's format it as a NOTE admonition. If it's required, let's number our steps and make this step 2.

This is an optional step provided as an example [here](https://docs.google.com/document/d/1uhpkq-crkshOUGL7OX9JiAXpd8UOmueMJJeeIGc_1aY/edit?tab=t.0). I will format it as a NOTE  to clarify that it is not necessary.

[id="enabling-guest-access-for-the-rbac-frontend-plugin_{context}"]
= Enabling guest access for the RBAC frontend plugin

> Use guest access for the Role-Based Access Control (RBAC) frontend plugin to test and develop, except in production. To enable the guest access for the RBAC frontend, configure the backend plugin and set up the guest authentication provider.

Reviewer: ```suggestion
Use guest access for the Role-Based Access Control (RBAC) frontend plugin to allow a user to test and develop code while restricting their access to the production environment. To enable the guest access for the RBAC frontend, configure the backend plugin and set up the guest authentication provider.
```
I'm actually not sure what we are trying to communicate - how do we state this more clearly while remaining technically accurate?

Reviewer: Can we also please state whether it is enabled or disabled by default. I assume disabled, but still good to clearly state that.

Author: > I'm actually not sure what we are trying to communicate - how do we state this more clearly while remaining technically accurate?

I agree that we need to maintain clarity. However, in my opinion, the suggestion changes the original meaning. I have made an adjustment to this part. Please let me know what you think. @PatAKnight, could you please provide more clarity on whether this is enabled or disabled by default?

Reviewer 2: The original ask was for some documentation on how to access the RBAC frontend plugin as a guest user. I assume that this is to test out the features of the RBAC plugin without the need to have an auth provider and other configuration values set up beforehand.

So maybe something along the lines of:
```
Use guest access with the Role-Based Access Control (RBAC) frontend plugin to allow a user to test role and policy creation without the need to set up and configure an authentication provider.
```

and then have a note that mentions not recommended for production scenarios.

.Prerequisites
* You have installed the `@janus-idp/backstage-plugin-rbac` plugin in {product-short}. For more information, see link:{plugins-configure-book-url}[{plugins-configure-book-title}].

.Procedure

To configure the RBAC backend plugin, complete the following steps:

> Update the `app-config.yaml` file to enable the permission framework as shown:

Reviewer: ```suggestion
. Update the `app-config.yaml` file to enable the permission framework as shown:
```
add numbering to an ordered list

> To configure the RBAC backend plugin, complete the following steps:
> 
> Update the `app-config.yaml` file to enable the permission framework as shown:

Reviewer: ```suggestion
To configure the RBAC backend plugin, complete the following step:

* Update the `app-config.yaml` file to enable the permission framework as shown:
```

nvm, it seems that this is a one-step procedure, so we use a bullet instead of a number

[source,yaml,subs=+quotes]
----
permission
  enabled: true
  rbac:
    admin:
      users:
        - name: user:default/guest
    pluginsWithPermission:
      - catalog
      - permission
      - scaffolder
----

[NOTE]
====
The `pluginsWithPermission` section of the `app-config.yaml` section includes only three plugins by default. Update the section as needed to include any additional plugins that also incorporate permissions.
====

> == Setting up the guest authentication provider

Reviewer: As we prepare to transition to DITA, it is increasingly important that we follow modularization guidelines. 
Each procedure should be contained in a single module (file), so we need to modularize this now or with a separate ticket / PR

Also determine the proper level offset

Author: Thank you, @linfraze. I have modularised the sections.

.Procedure

To enable guest authentication and to use it alongside the RBAC frontend plugin, complete the following step:

* In the `app-config.yaml` file, add the user entity reference to resolve and enable the `dangerouslyAllowOutsideDevelopment` option, as shown in the following example:

> 

Reviewer: formatted perfectly
